lib/que/metrics.rb

```diff
@@ -7,62 +7,32 @@
-      )
-      @worker_sleeping_seconds_total = register_counter(
-        :que_worker_sleeping_seconds_total, "Time spent sleeping, because no jobs"
-      )
-      @worker_running_seconds_total = register_counter(
-        :que_worker_running_seconds_total, "Time since starting to work jobs"
-      )
-      @worker_job_exists_total = register_counter(
-        :que_worker_job_exists__total, "Counter of number of attempts to check job existance in locking"
-      )
-      @worker_job_exists_seconds_total = register_counter(
-        :que_worker_job_exists_seconds_total, "Time spent checking if job exists as part of locking"
-      )
-      @worker_job_unlock_total = register_counter(
-        :que_worker_job_unlock_total, "Counter of number of advisory unlocks"
-      )
-      @worker_job_unlock_seconds_total = register_counter(
-        :que_worker_job_unlock_seconds_total, "Time spent unlocking advisory job locks"
-      )
-      @worker_job_acquire_total = register_counter(
-        :que_worker_job_acquire_total, "Counter of number of attempts to acquire jobs"
-      )
-      @worker_job_acquire_seconds_total = register_counter(
-        :que_worker_job_acquire_seconds_total, "Sum of time taken to acquire jobs"
-      )
+    def self.register_metrics(registry)
+      return registry if registry.exist?(:que_jobs_worked_total)
+
+      registry.instance_eval do
+        counter(:que_jobs_worked_total, "Counter for all jobs processed")
+        counter(:que_jobs_error_total, "Counter for all jobs that were run but errored")
+        counter(:que_jobs_worked_seconds_total, "Sum of the time taken processing each job class")
+        counter(:que_jobs_latency_seconds_total, "Sum of time spent by job and priority waiting in queue")
+        counter(:que_worker_sleeping_total, "Number of times worker slept due to no jobs")
+        counter(:que_worker_sleeping_seconds_total, "Time spent sleeping, because no jobs")
+        counter(:que_worker_running_seconds_total, "Time since starting to work jobs")
+        counter(:que_worker_job_exists__total, "Counter of number of attempts to check job existance in locking")
+        counter(:que_worker_job_exists_seconds_total, "Time spent checking if job exists as part of locking")
+        counter(:que_worker_job_unlock_total, "Counter of number of advisory unlocks")
+        counter(:que_worker_job_unlock_seconds_total, "Time spent unlocking advisory job locks")
+        counter(:que_worker_job_acquire_total, "Counter of number of attempts to acquire jobs")
+        counter(:que_worker_job_acquire_seconds_total, "Sum of time taken to acquire jobs")
+
+        self
+      end
     end
 
-    def expose(port: 8080)
+    # Start a webserver on the given port to expose Prometheus metrics. This should be
+    # given the registry that is used across all workers, to provide metrics for every
+    # worker thread.
+    def self.expose(registry, port: 8080)
```

Comment:
It makes more sense for this to be a method on the class, rather than the instance. While it exposes data from an instance, the server isn't necessarily related.